src/main/java/org/folio/consortia/service/impl/SyncPrimaryAffiliationServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package org.folio.consortia.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.consortia.client.SyncPrimaryAffiliationClient;
+import org.folio.consortia.domain.dto.SyncPrimaryAffiliationBody;
+import org.folio.consortia.domain.dto.SyncUser;
+import org.folio.consortia.domain.dto.User;
+import org.folio.consortia.service.SyncPrimaryAffiliationService;
+import org.folio.consortia.service.UserService;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class SyncPrimaryAffiliationServiceImpl implements SyncPrimaryAffiliationService {
+  private final UserService userService;
+  private final SyncPrimaryAffiliationClient syncPrimaryAffiliationClient;
+
+  @Override
+  @Async("asyncTaskExecutor")
+  public void syncPrimaryAffiliations(UUID consortiumId, String tenantId) {
+    log.info("Start creating user primary affiliation for tenant {}", tenantId);
+    List<User> users = new ArrayList<>();
+    try {
+      users = userService.getUsersByQuery("cql.allRecords=1", 0, Integer.MAX_VALUE);
+    } catch (Exception e) {
+      log.error("syncPrimaryAffiliations:: failed to retrieve users list");
```

Comment:
please add tenant id to this log, this endpoint to add tenant would be invoked simultaniosuly for 60+ tenants by pipeline, so we should know for each tenant it failed, and also need to add e to the log